core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -175,6 +175,29 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * Update the current threads class loader.
+   * Requires the current class loader is a MutableURLClassLoader, otherwise skips updating with a
+   * warning. Intended for use by addJar(), although constructing an instance of the class will
+   * still require:
+   * sc._jvm.java.lang.Thread.currentThread().getContextClassLoader().loadClass("class name")
+   * as described in SPARK-5185.
+   */
+  def updatePrimaryClassLoader(sc: SparkContext) {
```

Comment:
Ah thats a good point, it made a bit more sense when this was being done only for Python and there wasn't a flag. I'll switch it around.